PIL/TiffImagePlugin.py

```diff
@@ -225,6 +225,75 @@ def _limit_rational(val, max_val):
 _load_dispatch = {}
 _write_dispatch = {}
 
+class IFDRational(Fraction):
+    """ Implements a rational class where 0/0 is a legal value to match
+    the in the wild use of exif rationals.
+
+    e.g., DigitalZoomRatio - 0.00/0.00  indicates that no digital zoom was used
+    """
+
+    """ If the denominator is 0, store this as a float('nan'), otherwise store
+    as a fractions.Fraction(). Delegate as appropriate
+
+    """
+    
+    __slots__ = ('numerator', 'denominator', '_val') 
+
+    def __init__(self, value, denominator=1):
+        """
+        :param value: either an integer numerator, a
+        float/rational/other number, or an IFDRational
+        :param denominator: Optional integer denominator
+        """
+        self.denominator = denominator
+        self.numerator = value
+        
+        if type(value) == IFDRational:
+            self.denominator = value.denominator
+            self.numerator = value.numerator
+            self._val = value._val
+            return
+
+        if denominator == 0:
+            self._val = float('nan')
+            return
+
+        try:
+            if denominator == 1:
+                self._val = Fraction(value)
+            else:
+                self._val = Fraction(value, denominator)
```

Comment:
The problem with Py2.6 is that `Fraction` got a lot better in 2.7, and will take almost any number in the first position. In 2.6, you have to use the `Fraction.from_float` method (or similar)

Casting to an `int` there is going to return incorrect values.